### PR TITLE
add enable_storage_pools label to OperationErrorMetric

### PR DESCRIPTION
**Pull request by amacaskill**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

 /kind feature

> /kind flake

**What this PR does / why we need it**:
Adds enable_storage_pools label to the OperationErrorMetric so we can track CSI operation error rates when storage pools is used. 
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Adds enable_storage_pools label to the OperationErrorMetric so we can track CSI operation error rates when storage pools is used. 
```

### CODE REVIEW

File: pkg/metrics/metrics_test.go [hunk 1]
```diff
@@ -41,41 +42,69 @@ func CreateDiskWithConfidentialCompute(betaVersion bool, confidentialCompute boo
...
+		{
+			name:                              "test alphaDisk with storage pool projects/my-project/zone/us-central1-a/storagePools/sp1",
+			disk:                              CreateDiskWithStoragePool("projects/my-project/zone/us-central1-a/storagePools/sp1", hyperdiskBalanced),
+			expectedEnableConfidentialCompute: "false",
+			expectedDiskType:                  hyperdiskBalanced,
+			expectedEnableStoragePools:        "true",
+		},
+		{
+			name:                              "test alphaDisk with no storage pool",
+			disk:                              CreateDiskWithStoragePool("", hyperdiskBalanced),
```
Comment (sunnylovestiramisu) [inline pkg/metrics/metrics_test.go:90]: Can we add a test case that if the GetDisk call return an error(disk object will be empty), what is the expected output of this metrics?

Comment (amacaskill) [reply pkg/metrics/metrics_test.go:90]: Good call. Added.
